src/analysis/lattices/stack.h

```diff
@@ -258,7 +215,50 @@ template<Lattice L> class StackLattice {
     }
   }
 
-  Element getBottom() const noexcept { return Element{}; }
+  // When taking the LUB, we take the LUBs of the elements of each stack
+  // starting from the top of the stack. So, LUB([b, a], [b', a']) is
+  // [LUB(b, b'), LUB(a, a')]. If one stack is higher than the other,
+  // the bottom of the higher stack will be kept, while the LUB of the
+  // corresponding tops of each stack will be taken. For instance,
+  // LUB([d, c, b, a], [b', a']) is [d, c, LUB(b, b'), LUB(a, a')].
+  //
+  // We start at the top because it makes taking the LUB of stacks with
+  // different scope easier, as mentioned at the top of the file. It also
+  // fits with the conception of the stack starting at the top and having
+  // an infinite bottom, which allows stacks of different height and scope
+  // to be easily joined.
+  bool join(Element& self, const Element& other) const noexcept {
```

Comment:
Nit: Seems a bit weird to call the first element "self" as it implies the instance of the StackLattice class. Wdyt?

Reply:
Fair enough. Not sure what better to call them than `a` and `b` or something similarly unimaginative, though. Do you have ideas?

Reply:
The Obj-C dev in me quite liked the previous naming of `makeLeastUpperBound` because it encoded mutability at the function's call site. 

Since the first argument always has to match the LUB, maybe we can call it LUB?

Reply:
Hmm, the first argument is the LUB at the end of the method, but not necessarily at the beginning. It would be nice to have names that make sense both before and after.

Maybe `joinee` and `joiner`?

Reply:
Will do this globally as a follow-up.